src/account/Settings.tsx

```diff
@@ -324,10 +327,23 @@ export class Account extends React.Component<Props, State> {
     this.setState({ showAccountKeyModal: false })
   }
 
-  onPressContinueWithAccountRemoval = () => {
-    ValoraAnalytics.track(SettingsEvents.start_account_removal)
-    this.setState({ showAccountKeyModal: false })
-    this.props.navigation.navigate(Screens.BackupPhrase, { navigatedFromSettings: true })
+  onPressContinueWithAccountRemoval = async () => {
+    try {
+      this.setState({ showAccountKeyModal: false })
+      // Ugly hack to wait for the modal to close,
+      // otherwise the native modal PIN entry will not show up
+      // TODO: stop using ReactNative modals and switch to react-navigation modals
+      if (Platform.OS === 'ios') {
+        await sleep(500)
```

Comment:
just checking we still need this? is there some logic for chosing 500ms? i kind of worry that there is some race condition here and 500ms might not work for all users...

Reply:
We do still need this to allow the modal from reset account to fully dismiss before showing the pin page modal.

Reply:
> just checking we still need this? is there some logic for chosing 500ms? i kind of worry that there is some race condition here and 500ms might not work for all users...

Yes I hate this as well, it's enough time for the modal to close. We could improve this by trying to use the `onModalHide` but it requires more changes that I didn't want to do here.